src/psyclone/tests/parse/module_info_test.py

```diff
@@ -180,3 +175,120 @@ def test_mod_info_get_used_symbols_from_modules():
     used_symbols_cached = mod_info.get_used_symbols_from_modules()
     # The cached copy should be the same dictionary
     assert used_symbols_cached is used_symbols
+
+
+# -----------------------------------------------------------------------------
+@pytest.mark.usefixtures("change_into_tmpdir", "clear_module_manager_instance")
+def test_mod_info_get_psyir():
+    '''This tests the handling of PSyIR representation of the module.
+    '''
+
+    mod_man = ModuleManager.get()
+    dyn_path = get_base_path("dynamo0.3")
+    mod_man.add_search_path(f"{dyn_path}/driver_creation", recursive=False)
+
+    mod_info = mod_man.get_module_info("testkern_import_symbols_mod")
+    assert mod_info._psyir is None
+    psyir = mod_info.get_psyir()
+    assert isinstance(psyir, Container)
+    assert psyir.name == "testkern_import_symbols_mod"
+    # Make sure the PSyIR is cached:
+    assert mod_info._psyir.children[0] is psyir
+    # Test that we get the cached value (and not a new instance)
+    psyir_cached = mod_info.get_psyir()
+    assert psyir_cached is psyir
+
+    # Test that a file that can't be converted to PSyIR returns an
+    # empty FileContainer.
+    mod_man.add_search_path(dyn_path, recursive=False)
+    # The file 'broken_builtins_mod.f90' contains invalid Fortran and
+    # cannot be parsed:
+    broken_builtins = mod_man.get_module_info("broken_builtins_mod")
+    broken_builtins_psyir = broken_builtins.get_psyir()
+
+    # We should still get an empty Container with a dummy Container:
+    assert isinstance(broken_builtins_psyir, Container)
+    assert broken_builtins_psyir.name == "invalid-module"
+    assert isinstance(broken_builtins_psyir.parent, FileContainer)
+    assert broken_builtins_psyir.parent.name == "broken_builtins_mod.f90"
+
+
+# -----------------------------------------------------------------------------
+@pytest.mark.usefixtures("change_into_tmpdir", "clear_module_manager_instance",
+                         "mod_man_test_setup_directories")
+def test_generic_interface():
+    '''Tests that a generic interface works as expected.
+    '''
+    mod_man = ModuleManager.get()
+    mod_man.add_search_path("d1")
+    mod_man.add_search_path("d2")
+
+    mod_info = mod_man.get_module_info("g_mod")
+
+    # It should contain all three functions
+    assert mod_info.contains_routine("myfunc")
+    assert mod_info.contains_routine("myfunc1")
+    assert mod_info.contains_routine("myfunc2")
+
+    assert "myfunc" in mod_info._generic_interfaces
+
+
+# -----------------------------------------------------------------------------
+@pytest.mark.usefixtures("change_into_tmpdir", "clear_module_manager_instance",
+                         "mod_man_test_setup_directories")
+def test_resolve_routine():
```

Comment:
This needs a TODO as it will be removed/moved when Container has functionality to query for a Routine.

Reply:
Done.